ping.go

```diff
@@ -339,6 +339,58 @@ func (p *Pinger) Addr() string {
 	return p.addr
 }
 
+// SetSrcAddr set source by interface name or ipv4 address
+// Interface name can be "eth0" or ip like "192.168.1.239"
```

Comment:
The API feel confused here, imo. I think we should separate setting the source address to an IP literal vs. setting it to an address that's attached to a given interface.

Reply:
Are you suggesting to make 2 separate functions like, setSrcByIP and setScrByIface ?

Reply:
Unless there's a really compelling reason to combine them, I feel like 2 separate function is more intuitive, yes.